llvm/lib/CAS/OnDiskCommon.cpp

```diff
@@ -115,16 +116,25 @@ cas::ondisk::tryLockFileThreadSafe(int FD, std::chrono::milliseconds Timeout,
 }
 
 Expected<size_t> cas::ondisk::preallocateFileTail(int FD, size_t CurrentSize, size_t NewSize) {
-  auto CreateErrorFromErrno = [&]() -> Expected<size_t> {
-    std::error_code EC = errnoAsErrorCode();
+  auto CreateError = [&](std::error_code EC) -> Expected<size_t> {
     if (EC == std::errc::not_supported)
       // Ignore ENOTSUP in case the filesystem cannot preallocate.
       return NewSize;
+#if defined(HAVE_POSIX_FALLOCATE)
+    if (EC == std::errc::invalid_argument &&
+        CurrentSize < NewSize && // len > 0
+        NewSize < std::numeric_limits<off_t>::max()) // 0 <= offset, len < max
```

Comment:
>      [EINVAL]           The len argument was less than or equal to zero, the
>                         offset argument was less than zero, or the operation
>                         is not supported by the file system.

`CurrentSize` is in the offset position of the `posix_fallocate` call.
Shouldn't this check be verifying that `CurrentSize` is greater than or equal to zero?